packages/ptokens-swap/src/ptokens-swap-builder.ts

```diff
@@ -118,6 +125,55 @@ export class pTokensSwapBuilder {
     return this
   }
 
+  /**
+   * Set Interim Chain NetworkId.
+   * @param _networkId - The NetworkId of the Interim Chain.
+   * @returns The same builder. This allows methods chaining.
+   */
+  setInterimChain(_networkId: NetworkId) {
+    this._interimNetworkId = _networkId
+    return this
+  }
+
+  /**
+   * Set a Interim provider.
+   * @param provider - A pTokensEvmProvider.
+   * @returns The same builder. This allows methods chaining.
+   */
+  setInterimProvider(interimProvider: pTokensEvmProvider): this {
+    this._interimProvider = interimProvider
+    return this
+  }
+
+  /**
+   * Get interim chian hub address.
+   * @returns The interim hub address.
+   */
+  async getInterimHubAddress(networkId: NetworkId = INTERIM_CHAIN_NETWORK_ID): Promise<`0x${string}`> {
```

Comment:
This is a function I would move into ptokens-asset-evm/lib/index.ts as it is strictly related to evm.

```suggestion
  async getHubAddress(provider: pTokensEvmProvider, networkId: NetworkId): Promise<`0x${string}`> {
```

Reply:
And it can also be used in the builder for the pTokensEvmAsset. There is the same call to `hub`